### ISSUE TITLE
is there way to clear socket cache

### ISSUE THREAD

**Issue by hatemragab** [rikulo/socket.io-client-dart#156]
when I connect to socket and got disconnect because of network then emit("someThing")
when back online socket will automatically re Emit All 
I want to clean the cache if the socket state is reconnecting 
if use auto-connect false it works because I manually delete the old instance 
and create a new one 
but I want to use auto-connect true

thanks ❤

**Comment by hatemragab** [comment]
@jumperchen

**Comment by krunalpuri** [comment]
@hatemragab 
TLDR;
You can clear cache by add this line:
`IO.cache.clear();`

I had a similar issue where the socket would just cache the user token and eventually the token expires, put never updates.
Another issue was when a user logout and logs in with a new account, hence a new token. 
So I tried to calling `socket.dispose() `
which claims to destroy, close and disconnect the socket connection. 
However, I still had to call clear cache.

There are couple of other functions that might be useful based on your use-case:
`socket.off(EVENT, EVENTHANDLER) // unbinds a specific event listeners`
`socket.clearListeners() // unbinds all the listeners`
`socket.destroy // manually disconnect the socket`
`socket.disconnect or socket.close() // manually disconnect the socket`
Although, none of these seem to clear cache. 
I'm assuming this might be a design choice rather than a bug  🤔
May be someone from the dev team can explain it much better.

**Comment by hatemragab** [comment]
I hope that thanks for your response i will check io.cashe .delete

**Comment by hatemragab** [comment]
IO.cache.clear(); 
And socket.off(EVENT, EVENTHANDLER) for all events
Seem works fine for me now 
thanks

**Comment by hatemragab** [comment]
bro how to 

> @hatemragab
> TLDR;
> You can clear cache by adding these line:
> `import 'package:socket_io_client/socket_io_client.dart' as IO;`
> `IO.cache.clear();`
> 
> I had a similar issue where the socket would just cache the user token and eventually the token expires, put never updates.
> Another issue was when a user logout and logs in with a new account, hence a new token.
> So I tried to calling `socket.dispose() `
> which claims to destroy, close and disconnect the socket connection.
> However, I still had to call clear cache.
> 
> There are couple of other functions that might be useful based on your use-case:
> `socket.off(EVENT, EVENTHANDLER) // unbinds a specific event listeners`
> `socket.clearListeners() // unbinds all the listeners`
> `socket.destroy // manually disconnect the socket`
> `socket.disconnect or socket.close() // manually disconnect the socket`
> Although, none of these seem to clear cache.
> I'm assuming this might be a design choice rather than a bug 🤔
> May be someone from the dev team can explain it much better.

bro how to dismiss the socket if there error while connecting

**Comment by krunalpuri** [comment]
> bro how to
> 
> > @hatemragab
> > TLDR;
> > You can clear cache by adding these line:
> > `import 'package:socket_io_client/socket_io_client.dart' as IO;`
> > `IO.cache.clear();`
> > I had a similar issue where the socket would just cache the user token and eventually the token expires, put never updates.
> > Another issue was when a user logout and logs in with a new account, hence a new token.
> > So I tried to calling `socket.dispose() `
> > which claims to destroy, close and disconnect the socket connection.
> > However, I still had to call clear cache.
> > There are couple of other functions that might be useful based on your use-case:
> > `socket.off(EVENT, EVENTHANDLER) // unbinds a specific event listeners`
> > `socket.clearListeners() // unbinds all the listeners`
> > `socket.destroy // manually disconnect the socket`
> > `socket.disconnect or socket.close() // manually disconnect the socket`
> > Although, none of these seem to clear cache.
> > I'm assuming this might be a design choice rather than a bug 🤔
> > May be someone from the dev team can explain it much better.
> 
> bro how to dismiss the socket if there error while connecting

I'm not sure what you mean by "dismiss the socket".
You can register handlers when there is any error while connecting.
`socket.onConnectError(ConnectionErrorHandlerFunction)`
`socket.onConnectTimeout(ConnectionTimeoutHandlerFunction)`
`socket.onError(SocketErrorHandlerFunction)`

If you "auto-connect" enable, then it should be able to reconnect with the same socket without any issue.
